packages/@react-spectrum/table/src/Table.tsx

```diff
@@ -238,7 +238,8 @@ function TableCollectionView({layout, collection, focusedKey, renderView, render
     onVisibleRectChange(rect) {
       bodyRef.current.scrollTop = rect.y;
       setScrollLeft(bodyRef.current, direction, rect.x);
-    }
+    },
+    transitionDuration: collection.body.props.isLoading && collection.body.props.items?.length > 0 ? 0 : 500
```

Comment:
This will only work with dynamic tables. You should use `collection.size` instead.